simplified-books-database/src/main/java/org/nypl/simplified/books/book_database/BookDRMInformationHandleACS.kt

```diff
@@ -0,0 +1,200 @@
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * An information handle for Adobe ACS DRM.
+ */
+
+class BookDRMInformationHandleACS(
+  private val directory: File,
+  format: BookFormats.BookFormatDefinition,
+  private val onUpdate: () -> Unit
+) : BookDRMInformationHandle.ACSHandle(), BookDRMInformationHandleBase {
+
+  private val objectMapper = ObjectMapper()
+  private val closed = AtomicBoolean(false)
+
+  companion object {
+    fun nameRights(format: BookFormats.BookFormatDefinition): String =
+      "${format.shortName}-rights_adobe.xml"
+
+    fun nameACSM(format: BookFormats.BookFormatDefinition): String =
+      "${format.shortName}-meta_adobe.acsm"
+
+    fun nameMetaJSON(format: BookFormats.BookFormatDefinition) =
+      "${format.shortName}-meta_adobe.json"
+
+    fun names(format: BookFormats.BookFormatDefinition) =
+      listOf(
+        nameRights(format),
+        nameACSM(format),
+        nameMetaJSON(format)
+      )
+  }
+
+  init {
+    BookDRMInformationHandles.writeDRMInfo(
+      directory = this.directory,
+      format = format,
+      kind = BookDRMKind.ACS
+    )
+  }
+
+  private val fileAdobeRights: File =
+    File(this.directory, nameRights(format))
+  private val fileAdobeRightsTmp: File =
+    File(this.directory, nameRights(format) + ".tmp")
+  private val fileAdobeACSM: File =
+    File(this.directory, nameACSM(format))
+  private val fileAdobeACSMTmp: File =
+    File(this.directory, nameACSM(format) + ".tmp")
+  private val fileAdobeMeta: File =
+    File(this.directory, nameMetaJSON(format))
+  private val fileAdobeMetaTmp: File =
+    File(this.directory, nameMetaJSON(format) + ".tmp")
+
+  private val infoLock: Any = Any()
+  private var infoRef: BookDRMInformation.ACS =
+    synchronized(this.infoLock) {
+      this.loadInitial(
+        fileAdobeRights = this.fileAdobeRights,
+        fileAdobeACSM = this.fileAdobeACSM,
+        fileAdobeMeta = this.fileAdobeMeta
+      )
+    }
+
+  private fun loadInitial(
+    fileAdobeRights: File,
+    fileAdobeACSM: File,
+    fileAdobeMeta: File
+  ): BookDRMInformation.ACS {
+    return BookDRMInformation.ACS(
+      acsmFile = if (fileAdobeACSM.isFile) fileAdobeACSM else null,
+      rights = this.loadAdobeRightsInformationIfPresent(fileAdobeRights, fileAdobeMeta)
+    )
+  }
+
+  @Throws(IOException::class)
+  private fun loadAdobeRightsInformationIfPresent(
+    fileAdobeRights: File,
+    fileAdobeMeta: File
+  ): Pair<File, AdobeAdeptLoan>? {
+    return if (fileAdobeRights.isFile) {
+      this.loadAdobeRightsInformation(fileAdobeRights, fileAdobeMeta)
+    } else {
+      null
+    }
+  }
+
+  @Throws(IOException::class)
+  private fun loadAdobeRightsInformation(
+    fileAdobeRights: File,
+    fileAdobeMeta: File
+  ): Pair<File, AdobeAdeptLoan>? {
+    val serialized = FileUtilities.fileReadBytes(fileAdobeRights)
```

Comment:
You might consider using the extension functions from the Kotlin stdlib for some of these operations:

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/read-bytes.html

Reply:
Yep, can do. This was just copied and pasted from the original book format code.

Reply:
Ah! Well, if it was just copied over I don't care as much 😅